internal/storage/v1/cassandra/spanstore/dbmodel/model.go

```diff
@@ -53,6 +56,136 @@ type KeyValue struct {
 	ValueBinary  []byte  `cql:"value_binary"`
 }
 
+func (t *KeyValue) compareValues(that *KeyValue) int {
+	switch t.ValueType {
+	case stringType:
+		if t.ValueString != that.ValueString {
+			if t.ValueString < that.ValueString {
+				return -1
+			}
+			return 1
+		}
+	case boolType:
+		if t.ValueBool != that.ValueBool {
+			if !t.ValueBool {
+				return -1
+			}
+			return 1
+		}
+	case int64Type:
+		if t.ValueInt64 != that.ValueInt64 {
+			if t.ValueInt64 < that.ValueInt64 {
+				return -1
+			}
+			return 1
+		}
+	case float64Type:
+		if t.ValueFloat64 != that.ValueFloat64 {
+			if t.ValueFloat64 < that.ValueFloat64 {
+				return -1
+			}
+			return 1
+		}
+	case binaryType:
+		if c := bytes.Compare(t.ValueBinary, that.ValueBinary); c != 0 {
+			return c
+		}
+	default:
+		return -1 // theoretical case, not stating them equal but placing the base pointer before other
+	}
+	return 0
+}
+
+func (t *KeyValue) Compare(that any) int {
+	if that == nil {
+		if t == nil {
+			return 0
+		}
+		return 1
+	}
+	that1, ok := that.(*KeyValue)
+	if !ok {
+		that2, ok := that.(KeyValue)
+		if !ok {
+			return 1
+		}
+		that1 = &that2
+	}
+	if that1 == nil {
+		if t == nil {
+			return 0
+		}
+		return 1
+	} else if t == nil {
+		return -1
+	}
+	if t.Key != that1.Key {
+		if t.Key < that1.Key {
+			return -1
+		}
+		return 1
+	}
+	if t.ValueType != that1.ValueType {
+		if t.ValueType < that1.ValueType {
+			return -1
+		}
+		return 1
+	}
```

Comment:
if ValueType is numeric then

```suggestion
	if cmp := t.ValueType - that1.ValueType; cmp != 0 {
		return cmp
	}
```

Reply:
> _cmp(a, b) should return a negative number when a < b, a positive number when a > b and zero when a == b or a and b are incomparable in the sense of a strict weak ordering._

godocs explicitly provide this example:
```
sort.Slice(people, func(i, j int) bool { return people[i].Age < people[j].Age })
```